### ISSUE TITLE
Required Modules and Installation

### ISSUE THREAD

**Issue by joncodo** [friggframework/create-frigg-app#68]
# Overview

Right now, we create a bunch of different requirements in package.json for all the `frigg` related things.

```
"@friggframework/api-module-connectwise": "^0.8.26",
        "@friggframework/api-module-hubspot": "^0.8.24",
        "@friggframework/api-module-salesforce": "^0.8.30",
        "@friggframework/assertions": "^1.0.7",
        "@friggframework/core": "^0.2.11",
        "@friggframework/database": "^1.0.11",
        "@friggframework/errors": "^1.1.6",
        "@friggframework/integrations": "^1.0.22",
        "@friggframework/logs": "^1.0.9",
        "@friggframework/models": "^1.0.8",
        "@friggframework/module-plugin": "^1.0.24",
```

## Questions

1. I am not sure where the repos are for all of these but I have to assume that they all live within the `frigg` core project?
2. Would you like to just require `frigg` in the package.json?
3. What are the advantages and disadvantages that have been considered? 

I am not too opinionated here at all. I just want to document this decision for future users.

## Why this came up

I am just running into errors where it seems like the models package is not installing properly which I am sure I can fix.

**Comment by seanspeaks** [comment]
Yeah you've hit it on the head here...
1. The initial approach was to pull apart a base directory and create a number of small packages that are tested and fully extensible. The end goal was to combine those into a core package. I think it (probably) makes sense to have these packages all still get published individually but just require almost everything into core.

2. The api modules should all be packaged individually and installed individually... mostly because some of them will rely on external SDKs and having an app install all of _their_ dependencies when not using seems cumbersome. I can imagine some optionality here... make the whole library a dependency and just require in only what you need, then let webpack etc shake the tree out? Anywho.

3. The other area of concern are dependencies that are context dependent. We shouldn't make someone install the AWS SDK if they're using GCP; when we support non-MongoDB data store, we shouldn't make folks require in mongoose and other Mongo related packages; etc. I've considered basically calling those "core plugins".

So, for example, an AWS Frigg application would have @friggframework/core installed alongside @friggframework/plugin-aws and @friggframework/plugin-mongo and @friggframework/api-module-library (<= if we went the "import whole library" route)

Whereas GCP would have plugin-gcp etc.

**Comment by joncodo** [comment]
I totally agree. This will take some work but can be done.

> I am not sure where the repos are for all of these but I have to assume that they all live within the frigg core project?

Can you also answer this @seanspeaks ?

**Comment by seanspeaks** [comment]
Yup, friggframework/frigg packages for everything non API module library. api-module-library contains all of the api module packages.
